virtual/activations.go

```diff
@@ -382,6 +384,47 @@ func (a *activations) getServerState() (
 	return a.serverState.serverID, a.serverState.serverVersion
 }
 
+func (a *activations) close(ctx context.Context, numWorkers int) error {
+	log.Print("acquiring lock for closing actor activations")
+	a.Lock()
+	log.Print("acquired lock for closing actor activations")
+	defer a.Unlock()
+
+	sem := semaphore.NewWeighted(int64(numWorkers))
+
+	wg := sync.WaitGroup{}
+	closed := int64(0)
+	expected := int64(len(a._actors))
+	for actorId, futActor := range a._actors {
+		sem.Acquire(ctx, 1)
```

Comment:
I think you need to check the result of `sem.Acquire` here. If it failed, then `sem.Release()` later will probably cause a panic.